From 335f20ffc5b6676703f1b4ce7c6f4d3e0f5143b1 Mon Sep 17 00:00:00 2001
From: Pau Ruiz Safont <pau.ruizsafont@cloud.com>
Date: Mon, 27 Jan 2025 15:33:30 +0000
Subject: [PATCH 1/3] CA-399669: Do not exit with error when IPMI readings
 aren't available

This error made toolstack restarts fail.

Because not all drivers have ipmi available, exit gracefully instead.

Signed-off-by: Pau Ruiz Safont <pau.ruizsafont@cloud.com>
---
 ocaml/xcp-rrdd/bin/rrdp-dcmi/rrdp_dcmi.ml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ocaml/xcp-rrdd/bin/rrdp-dcmi/rrdp_dcmi.ml b/ocaml/xcp-rrdd/bin/rrdp-dcmi/rrdp_dcmi.ml
index 03afac48b..94824a86e 100644
--- a/ocaml/xcp-rrdd/bin/rrdp-dcmi/rrdp_dcmi.ml
+++ b/ocaml/xcp-rrdd/bin/rrdp-dcmi/rrdp_dcmi.ml
@@ -72,8 +72,8 @@ let _ =
   initialise () ;
   match discover () with
   | [] ->
-      D.info "IPMI DCMI power reading is unavailable" ;
-      exit 1
+      D.warn "IPMI DCMI power readings not available, stopping." ;
+      exit 0
   | _ ->
       D.info "IPMI DCMI power reading is available" ;
       main_loop ~neg_shift:0.5 ~target:(Reporter.Local 1)
-- 
2.39.5

